core/geometry/src/Geometry.ts

```diff
@@ -890,159 +936,207 @@ export class Geometry {
+     * For example, if "stepSize = 2" then we return "numSteps = 5" if 8 < total <= 10.
+     */
+    const numSteps = Math.floor((total + 0.999999 * stepSize) / stepSize);
+    if (numSteps < minCount)
       return minCount;
-    if (stepCount > maxCount)
+    if (numSteps > maxCount)
       return maxCount;
-    return stepCount;
+    return numSteps;
   }
   /**
-   * Test if x is in simple 0..1 interval. But optionally skip the test. (this odd behavior is very
-   * convenient for code that sometimes does not do the filtering.)
+   * Test if `x` is in simple 0..1 interval (but skip the test if `apply01 = false`).
+   * * This odd behavior is very convenient for code that sometimes does not do the filtering.
    * @param x value to test.
-   * @param apply01 if false, accept all x.
+   * @param apply01 if false, return `true` for all values of `x`.
    */
   public static isIn01(x: number, apply01: boolean = true): boolean {
     return apply01 ? x >= 0.0 && x <= 1.0 : true;
   }
   /**
-   * Test if x is in simple 0..1 interval. But optionally skip the test. (this odd behavior is very convenient
-   * for code that sometimes does not do the filtering.)
+   * Test if x is in simple 0..1 interval for a given positive `tolerance`.
+   * * Make sure to pass a positive `tolerance` because there is no check for that in the code.
    * @param x value to test.
-   * @param apply01 if false, accept all x.
+   * @param tolerance the tolerance.
    */
   public static isIn01WithTolerance(x: number, tolerance: number): boolean {
     return x + tolerance >= 0.0 && x - tolerance <= 1.0;
   }
   /**
-   * Restrict x so it is in the interval `[a,b]`, allowing a,b to be in either order.
-   * @param x
+   * Restrict x so it is in the interval `[a,b]` (allowing `a` and `b` to be in either order).
+   * @param x value to restrict
    * @param a (usually the lower) interval limit
    * @param b (usually the upper) interval limit
    */
   public static restrictToInterval(x: number, a: number, b: number): number {
     if (a <= b) {
-      if (x < a) return a;
-      if (x > b) return b;
+      if (x < a)
+        return a;
+      if (x > b)
+        return b;
       return x;
     }
-    // reversed interval ....
-    if (x < b) return b;
-    if (x > a) return a;
+    // reversed interval
+    if (x < b)
+      return b;
+    if (x > a)
+      return a;
     return x;
   }
   /**
    * Case-insensitive string comparison.
-   * * Return true if the toUpperCase values match.
+   * * Return `true` if the `toUpperCase` values of `string1` and `string2` match.
    */
   public static equalStringNoCase(string1: string, string2: string): boolean {
     return string1.toUpperCase() === string2.toUpperCase();
   }
-  /** Test for EXACT match of number arrays. */
+  /**
+   * Test for exact match of two number arrays.
+   * Returns `true` if both arrays have the same length and have the same entries (or both are empty arrays).
```

Comment:
Consider this change here and in similar variant methods below.
```suggestion
   * Returns `true` if both arrays have the same length and entries, or if both arrays are empty or undefined.
```